nibabel/ecat.py

```diff
@@ -155,10 +155,10 @@
 _dtdefs = ( # code, name, equivalent dtype
     (1, 'ECAT7_BYTE', np.uint8),
     (2, 'ECAT7_VAXI2', np.int16),
-    (3, 'ECAT7_VAXI4', np.float32),
+    (3, 'ECAT7_VAXI4', np.int32),
     (4, 'ECAT7_VAXR4', np.float32),
     (5, 'ECAT7_IEEER4', np.float32),
-    (6, 'ECAT7_SUNI2', np.uint16),
+    (6, 'ECAT7_SUNI2', np.int16),
```

Comment:
The only ECAT files I can find on our system have ECAT7_SUNI2 type, so very hard to know what the intention was.